longevity_test.py

```diff
@@ -35,35 +35,39 @@ class LongevityTest(ClusterTester):
     def _run_all_stress_cmds(self, stress_queue, params):
         stress_cmds = params['stress_cmd']
         if not isinstance(stress_cmds, list):
+            stress_cmds = [[stress_cmds]]
+        # Support k8s multi-tenant test (test runs on few clusters)
+        elif len(self.db_clusters_multitenant) == 1:
             stress_cmds = [stress_cmds]
         # In some cases we want the same stress_cmd to run several times (can be used with round_robin or not).
         stress_multiplier = self.params.get('stress_multiplier')
         if stress_multiplier > 1:
             stress_cmds *= stress_multiplier
 
-        for stress_cmd in stress_cmds:
-            params.update({'stress_cmd': stress_cmd})
-            self._parse_stress_cmd(stress_cmd, params)
-
-            # Run all stress commands
-            self.log.debug('stress cmd: {}'.format(stress_cmd))
-            if stress_cmd.startswith('scylla-bench'):
-                stress_queue.append(self.run_stress_thread(stress_cmd=stress_cmd,
-                                                           stats_aggregate_cmds=False,
-                                                           round_robin=self.params.get('round_robin')))
-            elif stress_cmd.startswith('cassandra-harry'):
-                stress_queue.append(self.run_stress_thread(**params))
-            else:
-                stress_queue.append(self.run_stress_thread(**params))
+        for tenant_idx, stress_cmds_set in enumerate(stress_cmds):
+            for stress_cmd in stress_cmds_set:
+                params.update({'stress_cmd': stress_cmd, "loaders_tenant": tenant_idx})
+                self._parse_stress_cmd(stress_cmd, params)
+
+                # Run all stress commands
+                self.log.debug('stress cmd: {}'.format(stress_cmd))
+                if stress_cmd.startswith('scylla-bench'):
+                    stress_queue.append(self.run_stress_thread(stress_cmd=stress_cmd,
```

Comment:
@vponomaryov how does the `run_stress_function` function should point the relevant cluster ? how did you did it in the performance tests class ? 

@juliayakovlev I think this is the issue, the stress is running on the first node only. since `run_stress_thread` is defaulting to `Tester.loader_set` and `Tester.db_cluster`

Reply:
@fruch due to the specifics of the SCT implementation where we have tight coupling to the single DB cluster I did following:
Each `ScyllaClusterStats` class inherits `PerformanceRegressionTest` one where `run_stress_thread` method refers to the proper DB cluster using old logic - `self.db_cluster` attribute.

And, for the last your statement - it `does not` default to Tester's `db_cluster` and `loader_set`.

Reply:
@juliayakovlev lets consider using the same approach @vponomaryov did in the perf tests, of inheritage of the test class. 

```python
class ScyllaClusterStats(LongevityTest):
     ...

class LongevityMulitTenantTest(LongevityTest):
     def setUp(self):
         # same as in PerformanceRegressionOperatorMultiTenantTest.setUp()

    def test_custom_time(self):
            for scylla_cluster_stats in self.scylla_clusters_stats:
                # run the test function in a thread using `ThreadPoolExecutor`
                scylla_cluster_stats.test_custom_time()
           # wait for all threads, fail on the first one failing.
```